src/abc_atlas_access/abc_atlas_cache/cloud_cache.py

```diff
@@ -1165,6 +1282,9 @@ def _download_file(self,
                                    f"{file_attributes}\n"
                                    "In {max_iter} iterations")
 
+            if file_attributes.local_path.exists():
```

Comment:
Aesthetically, should this be outside of the `while not self._file_exists(file_attributes)` block?

Reply:
No, because the method _file_exits checks for if it is in the list of downloaded files.